packages/bsky/src/migrate-moderation-data.ts

```diff
@@ -0,0 +1,280 @@
+              sql`NULL`.as('durationInHours'),
+              sql`NULL`.as('expiresAt'),
+              sql`json_build_object('reportType', "reasonType")`.as('meta'),
+              sql`id`.as('legacyRefId'),
+            ]),
+        )
+        .orderBy('createdAt', 'asc'),
+    )
+
+  await insertQuery.execute()
+  const totalEvents = await countEvents(db)
+  console.log(`Created ${totalEvents} events`)
+
+  return
+}
+
+const setReportedAtTimestamp = async (db: PrimaryDatabase) => {
+  console.log('Initiating lastReportedAt timestamp sync')
+  const didUpdate = await sql`
+    UPDATE moderation_subject_status
+    SET "lastReportedAt" = reports."createdAt"
+    FROM (
+      select "subjectDid", "subjectUri", MAX("createdAt") as "createdAt"
+      from moderation_report
+      where "subjectUri" is null
+      group by "subjectDid", "subjectUri"
+    ) as reports
+    WHERE reports."subjectDid" = moderation_subject_status."did"
+      AND "recordPath"=''
+  `.execute(db.db)
+
+  console.log(
+    `Updated lastReportedAt for ${didUpdate.numUpdatedOrDeletedRows} did subject`,
+  )
+
+  const contentUpdate = await sql`
+    UPDATE moderation_subject_status
+    SET "lastReportedAt" = reports."createdAt"
+    FROM (
+      select "subjectDid", "subjectUri", MAX("createdAt") as "createdAt"
+      from moderation_report
+      where "subjectUri" is not null
+      group by "subjectDid", "subjectUri"
+    ) as reports
+    WHERE reports."subjectDid" = moderation_subject_status."did"
+      AND "recordPath" is not null
+      AND POSITION(moderation_subject_status."recordPath" IN reports."subjectUri") > 0
+  `.execute(db.db)
+
+  console.log(
+    `Updated lastReportedAt for ${contentUpdate.numUpdatedOrDeletedRows} subject with uri`,
+  )
+}
+
+const createStatusFromActions = async (db: PrimaryDatabase) => {
+  const allEvents = await db.db
+    // @ts-ignore
+    .selectFrom('moderation_action')
+    // @ts-ignore
+    .where('reversedAt', 'is', null)
+    // @ts-ignore
+    .select((eb) => eb.fn.count<number>('id').as('count'))
+    .executeTakeFirstOrThrow()
+
+  const chunkSize = 2500
+  const totalChunks = Math.ceil(allEvents.count / chunkSize)
+
+  console.log(`Processing ${allEvents.count} actions in ${totalChunks} chunks`)
+
+  await db.transaction(async (tx) => {
+    let currentChunk = 1
+    let lastProcessedId = 0
+    do {
+      const eventsQuery = tx.db
+        // @ts-ignore
+        .selectFrom('moderation_action')
+        // @ts-ignore
+        .where('reversedAt', 'is', null)
+        // @ts-ignore
+        .where('id', '>', lastProcessedId)
+        .limit(chunkSize)
+        .selectAll()
+      const events = (await eventsQuery.execute()) as ModerationActionRow[]
+
+      for (const event of events) {
+        // Remap action to event data type
+        const actionParts = event.action.split('#')
+        await adjustModerationSubjectStatus(tx, {
+          ...event,
+          action: `${actionParts[0]}#modEvent${actionParts[1]
```

Comment:
There may be a few legacy actions where `actionParts[0]` is `com.atproto.admin.moderationAction` rather than `com.atproto.admin.defs`.  Since we know we want the first part to be `com.atproto.admin.defs`, perhaps we can hardcode it to that value.

Reply:
hmm... yeah I did see those even in the staging dump but thought we would wanna leave them alone.